modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/internal/TcpDiscoveryNode.java

```diff
@@ -631,6 +649,15 @@ public boolean isCacheClient() {
         clientRouterNodeId = U.readUuid(in);
     }
 
+    /**
+     * Return cluster region id.
+     *
+     * @return Cluster region id or null if not set.
+     */
+    public @Nullable Long getClusterRegionId() {
```

Comment:
1. First annotation, then modifiers
2. Why getClusterRegionId() is java.lang.Long but not primitive? Would not it be fine if it was 0 for cases if it is not set?